lectures/mle.md

```diff
@@ -3,12 +3,504 @@ jupytext:
+ax.set_xlim(-1,20)
+ax.hist(sample, density=True, bins=5_000, histtype='stepfilled', alpha=0.8)
+plt.show()
+```
+
+The histogram shows that many people have very low wealth and a few people have
+very high wealth.
+
+
+We will take the full population size to be
+
+```{code-cell} ipython3
+N = 100_000_000
+```
+
+How can we estimate total revenue from the full population using only the sample data?
+
+Our plan is to assume that wealth of each individual is a draw from a distribution with density $f$.
+
+If we obtain an estimate of $f$ we can then approximate $T$ as follows:
+
+$$
+    T = \sum_{i=1}^{N} h(w_i) 
+      = N \frac{1}{N} \sum_{i=1}^{N} h(w_i) 
+      \approx N \int_{0}^{\infty} h(w)f(w) dw
+$$ (eq:est_rev)
+
+(The sample mean should be close to the mean by the law of large numbers.)
+
+The problem now is: how do we estimate $f$?
+
+
+## Maximum Likelihood Estimation
+
+[Maximum likelihood estimation](https://en.wikipedia.org/wiki/Maximum_likelihood_estimation) 
+is a method of estimating an unknown distribution.
+
+Maximum likelihood estimation has two steps:
+
+1. Guess what the underlying distribution is (e.g., normal with mean $\mu$ and
+   standard deviation $\sigma$).
+2. Estimate the parameter values (e.g., estimate $\mu$ and $\sigma$ for the
+   normal distribution)
+
+One reasonable assumption for the wealth is that each
+that $w_i$ is [log-normally distributed](https://en.wikipedia.org/wiki/Log-normal_distribution),
+with parameters $\mu \in (-\infty,\infty)$ and $\sigma \in (0,\infty)$.
+
+This means that $\ln w_i$ is normally distributed with mean $\mu$ and
+   standard deviation $\sigma$.
+
+You can see that this is a reasonable assumption because if we histogram log wealth
+instead of wealth the picture starts to look something like a bell-shaped curve.
+
+```{code-cell} ipython3
+ln_sample = np.log(sample)
+fig, ax = plt.subplots()
+ax.hist(ln_sample, density=True, bins=200, histtype='stepfilled', alpha=0.8)
+plt.show()
+```
+
++++ {"user_expressions": []}
+
+Now our job is to obtain the maximum likelihood estimates of $\mu$ and $\sigma$, which
+we denote by $\hat{\mu}$ and $\hat{\sigma}$.
+
+These estimates can be found by maximizing the likelihood function given the
+data.
+
+The pdf of a lognormally distributed random variable $X$ is given by:
+$$
+f(x) = \frac{1}{x}\frac{1}{\sigma \sqrt{2\pi}} exp\left(\frac{-1}{2}\left(\frac{\ln x-\mu}{\sigma}\right)\right)^2
+$$
+
+Since $\ln X$ is normally distributed this is the same as
+$$
+f(x) = \frac{1}{x} \phi(x)
+$$
+where $\phi$ is the pdf of $\ln X$ which is normally distibuted with mean $\mu$ and variance $\sigma ^2$.
+
+For a sample $x = (x_1, x_2, \cdots, x_n)$ the _likelihood function_ is given by:
+$$
+\begin{aligned}
+L(\mu, \sigma | x_i) = \prod_{i=1}^{n} f(\mu, \sigma | x_i) \\
+L(\mu, \sigma | x_i) = \prod_{i=1}^{n} \frac{1}{x_i} \phi(\ln x_i)
+\end{aligned}
+$$
+
+Taking $\log$ on both sides gives us the _log likelihood function_ which is:
```

Comment:
@maanasee , just so you know, you need a blank line before and after $$.

Example: 

The function is

$$ 
 f(x) = 2x
$$

This function is differentiable...